src/main/kotlin/no/nav/familie/ef/sak/service/steg/BeregnYtelseSteg.kt

```diff
@@ -33,75 +37,102 @@ class BeregnYtelseSteg(private val tilkjentYtelseService: TilkjentYtelseService,
+                tilkjentYtelseService.opprettTilkjentYtelse(TilkjentYtelse(personident = aktivIdent,
+                                                                           behandlingId = behandling.id,
+                                                                           andelerTilkjentYtelse = nyeAndeler))
+            }
+            is Avslå -> {
+                feilHvis(behandling.type != BehandlingType.FØRSTEGANGSBEHANDLING) { "Kan kun avslå ved førstegangsbehandling" }
+            }
+            else -> {
             }
-
-            tilkjentYtelseService.opprettTilkjentYtelse(
-                    TilkjentYtelse(
-                            personident = aktivIdent,
-                            behandlingId = behandling.id,
-                            andelerTilkjentYtelse = nyeAndeler
-                    )
-            )
-        } else if (vedtak is Innvilget) {
-            error("Innvilget vedtak må ha minimum en beløpsperiode")
         }
+
         vedtakService.slettVedtakHvisFinnes(behandling.id)
         vedtakService.lagreVedtak(vedtakDto = vedtak, behandlingId = behandling.id)
         simuleringService.hentOgLagreSimuleringsresultat(behandling)
     }
 
+    private fun lagBeløpsperioderForInnvilgetVedtak(vedtak: Innvilget,
+                                                    behandling: Behandling,
+                                                    aktivIdent: String) =
+            beregningService.beregnYtelse(vedtak.perioder.tilPerioder(), vedtak.inntekter.tilInntektsperioder())
+                    .map { beløpsperiode ->
+                        AndelTilkjentYtelse(
+                                beløp = beløpsperiode.beløp.toInt(),
+                                stønadFom = beløpsperiode.periode.fradato,
+                                stønadTom = beløpsperiode.periode.tildato,
+                                kildeBehandlingId = behandling.id,
+                                personIdent = aktivIdent,
+                                samordningsfradrag = beløpsperiode.beregningsgrunnlag?.samordningsfradrag?.toInt() ?: 0,
+                                inntekt = beløpsperiode.beregningsgrunnlag?.inntekt?.toInt() ?: 0,
+                                inntektsreduksjon = beløpsperiode.beregningsgrunnlag?.avkortningPerMåned?.toInt() ?: 0,
+                        )
+                    }
+
     private fun andelerForRevurdering(behandling: Behandling,
                                       beløpsperioder: List<AndelTilkjentYtelse>): List<AndelTilkjentYtelse> {
+        val forrigeAndeler = hentForrigeTilkjenteYtelse(behandling)
+
+        return slåSammenAndelerSomSkalVidereføres(beløpsperioder, forrigeAndeler)
+    }
+
+    private fun andelerForOpphør(behandling: Behandling, opphørFom: LocalDate): List<AndelTilkjentYtelse> {
+        val forrigeAndeler = hentForrigeTilkjenteYtelse(behandling)
+
+        feilHvis(forrigeAndeler.andelerTilkjentYtelse.none { andel ->
```

Comment:
Er det mulig å sende inn opphør fra juni, hvis man har perioder som er: ? 
```
januar-februar
OG
september-november
```

Reply:
Det kan du prøve, men da feiler vi godt :) Da må du enten velge 

- `september` som ender med andel for `januar og februar`
- `februar` som ender med andel for `januar`

Reply:
👍 Vet egentlige ikke om det er ønskelig å kunne sende inn ett opphør i det caset når det ikke er i en periode. Det BURDE ju være i en periode